content/outcomes/cloud-native-development/specific__file_access.md

```diff
@@ -0,0 +1,15 @@
+---
+title: File Access
+weight: 160
+layout: single
+team:
+ - VMware Tanzu Labs
+---
+
+Applications running in VMware Tanzu Application Service live inside a temporary container with an ephemeral virtual machine, so you cannot depend on anything sitting around on the file system permanently. This is a good thing because dependencies on shared file access can induce unintentional shared state and coupling. Most of the time, we will look to refactor your application away from any dependence on the underlying filesystem, passing shared data around via web service calls, publish/subscribe events, or shared caches and databases. If file access is required (as an integration pattern with an external tool or team outside of your control), TAS provides “volume services” to map network folders to the twelve-factor concept of bound services. The [volume services documentation](https://docs.pivotal.io/application-service/2-11/devguide/services/using-vol-services.html) provides an overview, and a demo app, so try to run through that example to make sure volume services are available to you in your PCF environment.
```

Comment:
Aren't shared caches or databases just as bad as sharing a file system?

Reply:
"It depends." (c)

DB/Caches typically better scale as they have been designed to handle hundreds of clients at the same time with things like consistency, transactionality, high-availability, concurrency in mind from the beginning.

For simple use-cases, when handling very small data sets with arbitrary, probably unrelated data, file could be a better choice.

I slightly updated the formulation.